x/ratelimit/rate_limit.go

```diff
@@ -0,0 +1,155 @@
+package ratelimit
+
+import (
+	"encoding/json"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+
+	"github.com/cosmos/ibc-go/v3/modules/core/exported"
+
+	ratelimitkeeper "github.com/Stride-Labs/stride/v4/x/ratelimit/keeper"
+	"github.com/Stride-Labs/stride/v4/x/ratelimit/types"
+)
+
+// Parse the denom from the Send Packet that will be used by the rate limit module
+// The denom that the rate limiter will use for a SEND packet depends on whether
+//    it was a NATIVE token (e.g. ustrd, stuatom, etc.) or NON-NATIVE token (e.g. ibc/...)...
+//
+// We can identify if the token is native or not by parsing the trace denom from the packet
+// If the token is NATIVE, it will not have a prefix (e.g. ustrd),
+//    and if it is NON-NATIVE, it will have a prefix (e.g. transfer/channel-2/uosmo)
+//
+// For NATIVE denoms, return as is (e.g. ustrd)
+// For NON-NATIVE denoms, take the ibc hash (e.g. hash "transfer/channel-2/usoms" into "ibc/...")
+func ParseDenomFromSendPacket(packet transfertypes.FungibleTokenPacketData) (denom string) {
+	// Determine the denom by looking at the denom trace path
+	denomTrace := transfertypes.ParseDenomTrace(packet.Denom)
+
+	// Native assets will have an empty trace path and can be returned as is
+	if denomTrace.Path == "" {
```

Comment:
We should double-check this assumption - does this if-else block map onto the logic we think it does?
Path == "" => natively minted on chain
Path != "" => IBC token

Reply:
ibc-go makes the same assumption
https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/keeper/relay.go#L210-L215

Reply:
sounds good - might be worth adding a comment linking to that code